In [3]:
import numpy as np#linear algebra
import pandas as pd#data processing,csv file i/o(eg,pd.read_csv)

In [4]:
df=pd.read_csv("Churn_Modelling.csv")

FileNotFoundError: [Errno 2] No such file or directory: 'Churn_Modelling.csv'

In [5]:
df.shape

NameError: name 'df' is not defined

#### Total 10 thousands data points and 14 columns

In [ ]:
df.head()

In [ ]:
#rownumber customerId and surname are not useful,hence dropped
df.drop(columns =['RowNumber','CustomerId','Surname'],inplace=True)

In [ ]:
df.head()

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   CreditScore      10000 non-null  int64  
 1   Geography        10000 non-null  object 
 2   Gender           10000 non-null  object 
 3   Age              10000 non-null  int64  
 4   Tenure           10000 non-null  int64  
 5   Balance          10000 non-null  float64
 6   NumOfProducts    10000 non-null  int64  
 7   HasCrCard        10000 non-null  int64  
 8   IsActiveMember   10000 non-null  int64  
 9   EstimatedSalary  10000 non-null  float64
 10  Exited           10000 non-null  int64  
dtypes: float64(2), int64(7), object(2)
memory usage: 859.5+ KB


In [16]:
df.isnull()#there are no null values

CreditScore  Geography  Gender    Age  Tenure  Balance  NumOfProducts  \
0           False      False   False  False   False    False          False   
1           False      False   False  False   False    False          False   
2           False      False   False  False   False    False          False   
3           False      False   False  False   False    False          False   
4           False      False   False  False   False    False          False   
...           ...        ...     ...    ...     ...      ...            ...   
9995        False      False   False  False   False    False          False   
9996        False      False   False  False   False    False          False   
9997        False      False   False  False   False    False          False   
9998        False      False   False  False   False    False          False   
9999        False      False   False  False   False    False          False   

      HasCrCard  IsActiveMember  EstimatedSalary  Exited  
0         False           False            False   False  
1         False           False            False   False  
2         False           False            False   False  
3         False           False            False   False  
4         False           False            False   False  
...         ...             ...              ...     ...  
9995      False           False            False   False  
9996      False           False            False   False  
9997      False           False            False   False  
9998      False           False            False   False  
9999      False           False            False   False  

[10000 rows x 11 columns]

In [17]:
#gender and geography are object type we need to convert it into numerical form

In [18]:
df.duplicated().sum()#is there any duplicate value

0

In [21]:
#let us check how many customers have left the bank
df['Exited'].value_counts()
#7963 have left the bank here 0 was for left the bank

Exited
0    7963
1    2037
Name: count, dtype: int64

In [24]:
#let us check how many customers are from which geography
df['Geography'].value_counts()

Geography
France     5014
Germany    2509
Spain      2477
Name: count, dtype: int64

ther eare less customers from spain 2477
and more are form france 5014

#let us check how many males and how many females are there
df['Gender'].values_counts()

In [26]:
df['Gender'].value_counts()

Gender
Male      5457
Female    4543
Name: count, dtype: int64

In [28]:
#we want to convert the geography and gender which is a on=bject type convert into integer type
df=pd.get_dummies(df,columns=['Geography','Gender'],drop_first=True)

In [29]:
df.head()

CreditScore  Age  Tenure    Balance  NumOfProducts  HasCrCard  \
0          619   42       2       0.00              1          1   
1          608   41       1   83807.86              1          0   
2          502   42       8  159660.80              3          1   
3          699   39       1       0.00              2          0   
4          850   43       2  125510.82              1          1   

   IsActiveMember  EstimatedSalary  Exited  Geography_Germany  \
0               1        101348.88       1              False   
1               1        112542.58       0              False   
2               0        113931.57       1              False   
3               0         93826.63       0              False   
4               1         79084.10       0              False   

   Geography_Spain  Gender_Male  
0            False        False  
1             True        False  
2            False        False  
3            False        False  
4             True        False

In [30]:
#here geography and france are dropped and male column is dropped

In [39]:
#assign all features to x and target column to y
#divide the data into train and test
X=df.drop(columns=['Exited'])
y=df['Exited'].values

from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)
#20%data for testing and 80% for training

In [38]:
X

CreditScore  Age  Tenure    Balance  NumOfProducts  HasCrCard  \
0             619   42       2       0.00              1          1   
1             608   41       1   83807.86              1          0   
2             502   42       8  159660.80              3          1   
3             699   39       1       0.00              2          0   
4             850   43       2  125510.82              1          1   
...           ...  ...     ...        ...            ...        ...   
9995          771   39       5       0.00              2          1   
9996          516   35      10   57369.61              1          1   
9997          709   36       7       0.00              1          0   
9998          772   42       3   75075.31              2          1   
9999          792   28       4  130142.79              1          1   

      IsActiveMember  EstimatedSalary  Geography_Germany  Geography_Spain  \
0                  1        101348.88              False            False   
1                  1        112542.58              False             True   
2                  0        113931.57              False            False   
3                  0         93826.63              False            False   
4                  1         79084.10              False             True   
...              ...              ...                ...              ...   
9995               0         96270.64              False            False   
9996               1        101699.77              False            False   
9997               1         42085.58              False            False   
9998               0         92888.52               True            False   
9999               0         38190.78              False            False   

      Gender_Male  
0           False  
1           False  
2           False  
3           False  
4           False  
...           ...  
9995         True  
9996         True  
9997        False  
9998         True  
9999        False  

[10000 rows x 11 columns]

In [40]:
y

array([1, 0, 1, ..., 1, 1, 0], dtype=int64)

In [42]:
X_train.shape#there are 11 columns ie features

(8000, 11)

In [54]:
#let us make  uniform scaling of all features
from sklearn.preprocessing import StandardScaler
scaler=StandardScaler()
X_train_trf=scaler.fit_transform(X_train)
X_test_trf=scaler.transform(X_test)

# let us define ANN model

In [7]:
import tensorflow
from tensorflow import keras
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Dense

AttributeError: partially initialized module 'pandas' has no attribute '_pandas_parser_CAPI' (most likely due to a circular import)

In [ ]:
#Now let us define ANN model
model=Sequential()
model.add(Dense(3,activation='sigmoid',input_dim=11))
model.add(Dense(1,activation='sigmoid'))